### PR TITLE
Refs #32074 -- Fixed find_module()/find_loader() warnings on Python 3.10+.

### PR DESCRIPTION
**Pull request by felixxm**
Fixes deprecation warnings:
```
<frozen importlib._bootstrap_external>:1381: ImportWarning: TestFinder.find_spec() not found; falling back to find_module()
<frozen zipimport>:158: DeprecationWarning: zipimporter.find_module() is deprecated and slated for removal in Python 3.12; use find_spec() instead
<frozen zipimport>:122: DeprecationWarning: zipimporter.find_loader() is deprecated and slated for removal in Python 3.12; use find_spec() instead
```
ticket-32074

### CODE REVIEW

File: tests/utils_tests/test_module_loading.py [hunk 1]
```diff
@@ -188,28 +188,17 @@ class TestFinder:
...
-        return mod
+        return spec
 
 
 class CustomLoader(EggLoader):
     """The Custom Loader test is exactly the same as the EggLoader, but
-    it uses a custom defined Loader and Finder that is intentionally
-    split into two classes. Although the EggLoader combines both functions
-    into one class, this isn't required.
+    it uses a custom defined Loader class. Although the EggLoader combines both
```
Comment (ngnpope) [inline tests/utils_tests/test_module_loading.py:200]: An aside: I see the reference to `EggLoader`. Haven't Python "eggs" been pretty much obsolete for ages? Can we remove this or the bits that are related to "eggs"? (Something to create a separate ticket for, if so.)

File: tests/utils_tests/test_module_loading.py [hunk 2]
```diff
@@ -188,28 +188,17 @@ class TestFinder:
...
-
-class TestLoader:
-    def __init__(self, importer):
-        self.importer = importer
-
-    def load_module(self, name):
-        mod = self.importer.load_module(name)
-        mod.__loader__ = self
-        return mod
+        return spec
```
Comment (ngnpope) [inline tests/utils_tests/test_module_loading.py:195]: This following is just the same as `return spec`:
```python
        if spec is None:
            return
        return spec
```
So:
```python
    def find_spec(self, path, target=None):
        return self.importer.find_spec(path, target)
```

**Review by ngnpope — APPROVED** [review]
Minor nit, but LGTM.

File: tests/utils_tests/test_module_loading.py (hunk 2 above)
Comment (felixxm) [reply tests/utils_tests/test_module_loading.py]: Updated.

File: tests/utils_tests/test_module_loading.py (hunk 1 above)
Comment (felixxm) [reply tests/utils_tests/test_module_loading.py:218]: Yes, it's a separate thing.

**Comment by felixxm** [comment]
@ngnpope Thanks for checking :+1:

**Pull request merged**